src/global_distrib/mod_global_distrib_mapping.erl

```diff
@@ -298,3 +302,14 @@ do_lookup_jid({_, _, _} = Jid) ->
                 BareJid -> ets_cache:lookup(?JID_TAB, BinJid, LookupInDB(jid:to_binary(BareJid)))
             end
     end.
+
+-spec get_advertised_endpoints(Opts :: list()) -> [endpoint()].
+get_advertised_endpoints(Opts) ->
+    Conns = proplists:get_value(connections, Opts, []),
+    AdvEndps = proplists:get_value(advertised_endpoints, Conns),
+    case AdvEndps of
+        undefined ->
```

Comment:
Probably easier to just return `proplists:get_value(advertised_endpoints, Conns, false)`